Lib/asyncio/tasks.py

```diff
@@ -678,21 +675,17 @@ def _ensure_future(coro_or_future, *, loop=None):
     try:
         return loop.create_task(coro_or_future)
     except RuntimeError:
-        if not called_wrap_awaitable:
-            coro_or_future.close()
+        coro_or_future.close()
         raise
 
 
-@types.coroutine
-def _wrap_awaitable(awaitable):
+async def _wrap_awaitable(awaitable):
     """Helper for asyncio.ensure_future().
 
     Wraps awaitable (an object with __await__) into a coroutine
     that will later be wrapped in a Task by ensure_future().
     """
-    return (yield from awaitable.__await__())
-
-_wrap_awaitable._is_coroutine = _is_coroutine
+    return await awaitable
```

Comment:
this will insist on objects being real awaitables rather than virtual awaitables - eg:

```python
import asyncio.tasks
import collections.abc
import unittest.mock


@collections.abc.Awaitable.register
class sleep_0:
    def __init__(self):
        self.__await__ = lambda: iter((None, ))


async def _wrap_awaitable(awaitable):
    return await awaitable


async def amain():
    await asyncio.ensure_future(sleep_0())  # this works
    with unittest.mock.patch("asyncio.tasks._wrap_awaitable", new=_wrap_awaitable):
        await asyncio.ensure_future(sleep_0())  # this will break

asyncio.run(amain())
```